app/helpers/toolbar_helper.rb

```diff
@@ -191,19 +191,20 @@ def toolbar_button_normal(props, view_button = false)
   # attributes in html
   #
   def data_hash_keys(props)
-    hsh = %i(pressed popup console_url prompt explorer onwhen url_parms url).each_with_object({}) do |key, h|
-      h["data-#{key}"] = props[key] if props.key?(key)
+    %i(pressed popup console_url prompt explorer onwhen url_parms url).each_with_object({}) do |key, h|
+      h[key] = props[key] if props.key?(key)
     end
-    hsh
   end
 
   # Calculate common html tag keys and values from toolbar button definition
   #
   def prepare_tag_keys(props)
-    h = data_hash_keys(props)
+    h = {
+      'data'       => data_hash_keys(props),
```

Comment:
Nice, but we need to make doubly sure none of the attributes from `data_hash_keys` are ever set via `data-something`...

`= text_field_tag("foo", "bar", :data => {'foo' => 5, "bar" => "666"}, "data-baz" => "borg", "data-foo" => 6)`

yields

`<input type="text" name="foo" id="foo" value="bar" data-foo="5" data-bar="666" data-baz="borg">`

(that is, the value passed to `data` overrides the value passed to `data-foo`)